cli/src/cmd/cast/create2.rs

```diff
@@ -0,0 +1,274 @@
+    ends_with: Option<String>,
+    #[clap(long, short, help = "Sequence that the address has to match", value_name = "HEX")]
+    matching: Option<String>,
+    #[clap(short, long)]
+    case_sensitive: bool,
+    #[clap(
+        short,
+        long,
+        help = "Address of the contract deployer.",
+        default_value = "0x4e59b44847b379578588920ca78fbf26c0b4956c",
+        value_name = "ADDRESS"
+    )]
+    deployer: Address,
+    #[clap(
+        short,
+        long,
+        help = "Init code of the contract to be deployed.",
+        value_name = "HEX",
+        default_value = ""
+    )]
+    init_code: String,
+    #[clap(
+        alias = "ch",
+        long,
+        help = "Init code hash the contract to be deployed.",
+        value_name = "HEX"
+    )]
+    init_code_hash: Option<String>,
+}
+
+#[allow(dead_code)]
+pub struct Create2Output {
+    address: Address,
+    salt: U256,
+}
+
+impl Cmd for Create2Args {
+    type Output = Create2Output;
+
+    fn run(self) -> eyre::Result<Self::Output> {
+        Create2Args::generate_address(self)
+    }
+}
+
+impl Create2Args {
+    fn generate_address(self) -> Result<Create2Output> {
+        let Create2Args {
+            starts_with,
+            ends_with,
+            matching,
+            case_sensitive,
+            deployer,
+            init_code,
+            init_code_hash,
+        } = self;
+
+        let mut regexs = vec![];
+
+        if let Some(mut matches) = matching {
+            if starts_with.is_some() || ends_with.is_some() {
+                eyre::bail!("Either use --matching or --starts/ends-with");
+            }
+
+            if matches.len() != 40 {
+                if !(matches.len() == 42 && matches.starts_with("0x")) {
+                    eyre::bail!("Please provide a 40 characters long sequence for matching");
+                }
+                matches = matches.strip_prefix("0x").unwrap().to_string();
+            }
+
+            hex::decode(&matches.replace('X', "0")).wrap_err("invalid matching hex provided")?;
+            regexs.push(matches.replace('X', "."));
+        }
+
+        if let Some(prefix) = starts_with {
+            let pad_width = prefix.len() + prefix.len() % 2;
+            hex::decode(format!("{:0>width$}", prefix, width = pad_width))
+                .wrap_err("invalid prefix hex provided")?;
+            regexs.push(format!(r"^{}", prefix));
+        }
+        if let Some(suffix) = ends_with {
+            let pad_width = suffix.len() + suffix.len() % 2;
+            hex::decode(format!("{:0>width$}", suffix, width = pad_width))
+                .wrap_err("invalid suffix hex provided")?;
+            regexs.push(format!(r"{}$", suffix));
+        }
+
+        assert!(
+            regexs.iter().map(|p| p.len() - 1).sum::<usize>() <= 40,
+            "vanity patterns length exceeded. cannot be more than 40 characters",
+        );
```

Comment:
is this a sanity `assert!` or something that depends on user input?

for former, can do a `debug_assert!` for latter, handle this as error.

Reply:
`debug_assert!` seems also to `panic!`, I guess that I should use `eyre::bail!` here.